sources/packages/backend/apps/api/src/route-controllers/student-account-applications/_test_/e2e/student-account-application.aest.controller.approveStudentAccountApplication.e2e-spec.ts

```diff
@@ -0,0 +1,348 @@
+    const studentAccountApplication = await db.studentAccountApplication.save(
+      createFakeStudentAccountApplication(
+        { user },
+        { initialValues: { submittedData } },
+      ),
+    );
+
+    await saveFakeSFASIndividual(db.dataSource, {
+      initialValues: {
+        lastName: user.lastName,
+        birthDate: TEST_BIRTH_DATE1,
+        sin: TEST_SIN2,
+      },
+    });
+
+    const endpoint = `/aest/student-account-application/${studentAccountApplication.id}/approve`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+    // Mock the form.io response.
+    sharedFormService.dryRunSubmission = jest.fn().mockResolvedValue({
+      valid: true,
+      formName: FormNames.StudentProfile,
+      data: { data: submittedData },
+    });
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(submittedData)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        expect(response.body.id).toBeGreaterThan(0);
+      });
+    // Check that the notification is in the database.
+    const notification = await db.notification.findOne({
+      select: {
+        id: true,
+        dateSent: true,
+        messagePayload: true,
+        notificationMessage: { templateId: true },
+      },
+      relations: { notificationMessage: true, user: true },
+      where: {
+        dateSent: IsNull(),
+        notificationMessage: {
+          id: NotificationMessageType.PartialStudentMatchNotification,
+        },
+      },
+    });
+
+    expect(notification.messagePayload["email_address"]).toEqual(
+      "dummy@some.domain",
+    );
+
+    expect(notification.messagePayload["personalisation"]).toEqual({
+      lastName: user.lastName,
+      givenNames: user.firstName,
+      studentEmail: user.email,
+      birthDate: "Jan 31 2001",
+      matches: "Last name and birth date match.",
+      matchTime: expect.any(String),
+    });
+  });
+
+  it("Should send a notification message when at least a partial match is found with matching last name and SINs for importing a student record from SFAS.", async () => {
+    // Arrange
+    const user = await db.user.save(createFakeUser());
+    // Submitted data to simulated the exists student account already saved on DB.
+    // Same data will be used to be submitted when Ministry is approving it.
+    const submittedData = {
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+      dateOfBirth: TEST_BIRTH_DATE1,
+      phone: "1234567890",
+      sinNumber: TEST_SIN1,
+      mode: "aest-account-approval",
+      identityProvider: "bceidboth",
+      sinConsent: true,
+      gender: "X",
+      addressLine1: "address 1",
+      city: "Victoria",
+      country: "Canada",
+      postalCode: "H1H1H1H",
+      provinceState: "BC",
+      selectedCountry: "canada",
+      canadaPostalCode: "H1H1H1H",
+    };
```

Comment:
If the payload is pretty similar across all the methods, it can be centralized in a common method in this file.
As a suggestion, something as below.
```ts
function createFakeSubmittedData(
    user: Pick<User, "firstName" | "lastName" | "email">,
    options?: {
      dateOfBirth: Date;
      sin?: string;
    },
  ): unknown {
    return {
      firstName: user.firstName,
      lastName: user.lastName,
      email: user.email,
      dateOfBirth: options?.dateOfBirth ?? new Date(),
      phone: options?.sin ?? "1234567890",
      sinNumber: TEST_SIN1,
      mode: "aest-account-approval",
      identityProvider: "bceidboth",
      sinConsent: true,
      gender: "X",
      addressLine1: "address 1",
      city: "Victoria",
      country: "Canada",
      postalCode: "H1H1H1H",
      provinceState: "BC",
      selectedCountry: "canada",
      canadaPostalCode: "H1H1H1H",
    };
  }
```

Reply:
updated